Core/src/ExponentialBackoff.php

```diff
@@ -49,20 +49,34 @@ class ExponentialBackoff
      */
     private $onRetryException;
 
+    /**
+     * @var callable|null
+     */
+    private $onExecutionStart;
+
     /**
      * @param int $retries [optional] Number of retries for a failed request.
      * @param callable $retryFunction [optional] returns bool for whether or not to retry
      * @param callable $onRetryException [optional] runs before the $retryFunction. Unlike the $retryFunction,
-     * this function isn't responsible to decide if a retry should happen or not,
-     * but it gives the users flexibility to consume exception messages and add custom logic.
-     * Ex: One might want to change heaers on every retry, this function can be used to achieve
-     * such a functionality.
+     *   this function isn't responsible to decide if a retry should happen or not,
+     *   but it gives the users flexibility to consume exception messages and add custom logic.
+     *   Ex: One might want to change heaers on every retry, this function can be used to achieve
+     *   such a functionality.
+     * @param callable $onExecutionStart [optional] runs before execution of the
+     *   execute function. Taken in $arguments as reference and thus gives users,
+     *   the flexibility to add custom logic before the execution of request starts.
```

Comment:
Since this function can override the headers/arguments which were also passed in `$options`, we should mention this in the doc string.

Reply:
Okay, explicitly mentioned this in this commit[72a5c27](https://github.com/googleapis/google-cloud-php/pull/5741/commits/72a5c2751e7d6ad58737fc499852cb4e9e64ab6c)

Thanks for suggesting